internal/release/main.go

```diff
@@ -0,0 +1,218 @@
+	repoPath, err := getProjectPath()
+	if err != nil {
+		return nil, err
+	}
+	r, err := git.PlainOpen(repoPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get repo: %w", err)
+	}
+
+	latestTag := fmt.Sprintf("%s/%s", module, lastTagVersion.Original())
+	latestTagRef, err := r.Tag(latestTag)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get tag: %w", err)
+	}
+
+	tagCommit, err := r.CommitObject(latestTagRef.Hash())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get commit from tag: %w", err)
+	}
+
+	headRef, err := r.Head()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get reference: %w", err)
+	}
+
+	headCommit, err := r.CommitObject(headRef.Hash())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get commit: %w", err)
+	}
+
+	pathPrefix := fmt.Sprintf("%s/", module)
+	commits, err := r.Log(&git.LogOptions{
+		From: headCommit.Hash,
+		PathFilter: func(path string) bool {
+			return strings.HasPrefix(path, pathPrefix)
+		},
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get commits: %w", err)
+	}
+
+	var commitMessages []string
+	// These commits are in the order of most recent first
+	err = commits.ForEach(func(c *object.Commit) error {
+		if c.Hash == tagCommit.Hash {
+			// Once we reach the tag's commit, stop iterating
+			return storer.ErrStop
+		}
+		commitMessages = append(commitMessages, c.Message)
+		return nil
+	})
+
+	if err != nil && !errors.Is(err, storer.ErrStop) {
+		return nil, fmt.Errorf("could not iterate over commits %w", err)
+	}
+
+	return commitMessages, nil
+}
+
+func getModuleTags(module string) ([]string, error) {
+	repoPath, err := getProjectPath()
+	if err != nil {
+		return nil, err
+	}
+	r, err := git.PlainOpen(repoPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open repository: %w", err)
+	}
+
+	tagRefs, err := r.Tags()
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch tags: %w", err)
+	}
+
+	var filteredTags []string
+	tagPrefix := fmt.Sprintf("refs/tags/%s/", module)
+	err = tagRefs.ForEach(func(ref *plumbing.Reference) error {
+		if strings.HasPrefix(string(ref.Name()), tagPrefix) {
+			filteredTags = append(filteredTags, strings.TrimPrefix(string(ref.Name()), tagPrefix))
+		}
+		return nil
+	})
+
+	return filteredTags, err
+}
+
+const (
+	major = "major"
+	minor = "minor"
+	patch = "patch"
+)
+
+func getTypeOfChange(commits []string) string {
+	// https://regex101.com/r/obSlh6/1
+	// Regex for conventional commits
+	commitRegex := regexp.MustCompile(`^(\w+)(\([\w\-.]+\))?(!)?:(\s+.*)`)
+	category := patch
+	for _, commit := range commits {
+		matches := commitRegex.FindStringSubmatch(commit)
+		if matches != nil {
+			commitType := matches[1]
+			isBreaking := matches[3] == "!"
+
+			if isBreaking {
+				category = major
+				break
+			} else if commitType == "feat" {
+				category = minor
+			}
+
+		}
+	}
+	return category
+}
+
+func main() {
+	module := os.Args[1]
```

Comment:
wouldn't hurt to add a check to ensure a single argument is passed to the program

`if len(os.Args) != 2`